### PR TITLE
Athletic Fishing Gloves and Fishing Module to fish without a fishing rod

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This PR adds a pair of fishing gloves that let you fish and work out at once, and also a MODsuit module that lets you use MOD gloves to do the same thing (without the workout, an external fishing rod has to be inserted first). In both cases, you can equip or unequip bait, hook and line by using the fishing rod interface which you can open by right-clicking the gloves. To get both of them, you've to perform the first fish scanning experiment.

I had to refactor the profound fisher component a bit to do this.

## Why It's Good For The Game
Interweaving a few different features with fishing. Fishing and Athletics are both skills, so I thought it'd be nice if it were a way to take advantage of both, and level them up accordingly. Also fishing gloves with maxxed out athletics can make fishing a lot noticeably easier at higher difficulty.

## Changelog

:cl:
add: Added Athletic Fishing Gloves and Fishing Glove Module to the advanced fishing tech node. Both can be used to fish without having to hold a fishing rod.  The athletic fishing gloves will also train your athletics skill.
/:cl:

### CODE REVIEW

**Comment by HoodMilk** [comment]
Barehanded fishing should cause some damage to your arm based on the weight of the fish for the true redneck noodling experience.

**Comment by Ghommie** [comment]
Ok.

**Comment by CRITAWAKETS** [comment]
oh fuck yeah we've unlocked barbarian fishing now

File: code/modules/fishing/fishing_minigame.dm [hunk 1]
```diff
@@ -196,21 +196,25 @@
...
-		RegisterSignal(user, COMSIG_MOVABLE_MOVED, PROC_REF(on_user_move))
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, PROC_REF(on_lure_or_user_move))
+		RegisterSignal(lure, COMSIG_MOVABLE_MOVED, PROC_REF(on_lure_or_user_move))
+		RegisterSignal(user, SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), PROC_REF(on_hands_blocked))
+	RegisterSignal(user, SIGNAL_REMOVETRAIT(TRAIT_PROFOUND_FISHER), PROC_REF(no_longer_fishing))
 	active_effects = bitfield_to_list(special_effects & FISHING_MINIGAME_ACTIVE_EFFECTS)
 	// If fishing line breaks los / rod gets dropped / deleted
 	RegisterSignal(used_rod, COMSIG_ITEM_ATTACK_SELF, PROC_REF(on_attack_self))
-	ADD_TRAIT(user, TRAIT_GONE_FISHING, REF(src))
+	ADD_TRAIT(user, TRAIT_GONE_FISHING, WEAKREF(src))
```
Comment (Jacquerel) [inline code/modules/fishing/fishing_minigame.dm:217]: Can you use a weakref as a trait source? That's news to me.
...and should you?

Comment (Ghommie) [reply code/modules/fishing/fishing_minigame.dm:217]: Apparently yes. There are lot of things that use weakrefs as sources.

**Pull request merged**